pkg/results/reader.go

```diff
@@ -155,8 +156,20 @@ func ExtractIntoStruct(predicate func(string) bool, path string, info os.FileInf
 		if !ok {
 			return errors.New("info.Sys() is not a reader")
 		}
+		// TODO(chuckha) there must be a better way
```

Comment:
upstream has the "universal decoder", I'll have to dig up a link.

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/util.go#L244 

legacyscheme.Codecs.UniversalDecoder()

Reply:
i looked at it briefly but it looked like it was pretty k8s specific. Maybe worth looking at deeper though.

For now, I went with the simplest approach possible.

Reply:
UniversalDecoder only decodes what kube's apimachinery can handle. That's usually json, yaml, and proto.